includes/data-port/class-sensei-import-csv-reader.php

```diff
@@ -60,21 +60,94 @@ class Sensei_Import_CSV_Reader {
 	public function __construct( $csv_file, $completed_lines = 0, $lines_per_batch = 30 ) {
 		$this->file = new SplFileObject( $csv_file );
 		$this->file->setFlags( SplFileObject::READ_CSV );
+		$this->set_delimiter();
 
+		$this->file->seek( PHP_INT_MAX );
+		$this->total_lines     = $this->file->key();
+		$this->completed_lines = $completed_lines;
+		$this->is_completed    = $this->completed_lines >= $this->total_lines;
+		$this->lines_per_batch = $lines_per_batch;
+	}
+
+	/**
+	 * Set the delimiter to read the CSV file.
+	 *
+	 * The delimiter detection works testing the delimiter which find more columns.
+	 */
+	private function set_delimiter() {
```

Comment:
Maybe this could be named something like `detect_delimiter`?

Reply:
Good! Updated here: https://github.com/Automattic/sensei/pull/3364/commits/7cd461bc0c93b5888df925ca7c44c6a527e50cc3